neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -128,32 +131,68 @@ class NeuralynxRawIO(BaseRawIO):
         ("channel_id", "uint32"),
         ("sample_rate", "uint32"),
         ("nb_valid", "uint32"),
-        ("samples", "int16", (NcsSection._RECORD_SIZE)),
+        ("samples", "int16", NcsSection._RECORD_SIZE),
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
+            include_filenames=None,
+            exclude_filenames=None,
+            keep_original_times=False,
+            strict_gap_mode=True,
+            filename=None,
+            exclude_filename=None,
+            **kargs
     ):
 
-        if dirname != "":
-            self.dirname = dirname
-            self.rawmode = "one-dir"
-        elif filename != "":
-            self.filename = filename
-            self.rawmode = "one-file"
+        if not dirname:
+            raise ValueError("`dirname` cannot be empty.")
+
+        if filename:
+            include_filenames = filename
```

Comment:
filename implies singular so we would expect the user to only give one filename. In this case we would want to convert that single file name to a list.

Reply:
This is updated. But as `include_filenames` supports both single and multiple files. So it should make not difference.

Reply:
I'm wondering if we should make `include_filenames` be an iterable. Such that filename needs to be converted. But I could be convinced otherwise. I'm of the school of thought that we should be more limited in what we accept so that we can reduce our checks on functions, but others on the team like being as accommodating as possible so we could accept both. Let's think about this a bit.